kolibri/plugins/coach/assets/src/views/learners/LearnerSummaryPage/LearnerHeader.vue

```diff
@@ -182,13 +184,64 @@
         };
       },
     },
+    methods: {
+      exportCSV() {
+        let filteredExams = this.exams.filter(exam =>
+          this.getLearnersForExam(exam).includes(this.learner.id),
+        );
+
+        let filteredLessons = this.lessons.filter(lesson =>
+          this.getLearnersForLesson(lesson).includes(this.learner.id),
+        );
+
+        filteredLessons = filteredLessons.map(lesson => {
+          lesson.status = this.getLessonStatusStringForLearner(lesson.id, this.learner.id);
+          return lesson;
+        });
+
+        filteredExams = filteredExams.map(exam => {
+          exam.statusObj = this.getExamStatusObjForLearner(exam.id, this.learner.id);
+          return exam;
+        });
```

Comment:
Is there a particular reason for making and mutating the `filtered*` variables rather than chaining the functions like `const filteredExams = this.exams.filter(...).map(...)`?

Reply:
Ahh yes this can fixed further. Are you suggesting something like this?
```
        const filteredExams = this.exams
          .filter(exam => this.getLearnersForExam(exam).includes(this.learner.id))
          .map(exam => ({
            ...exam,
            statusObj: this.getExamStatusObjForLearner(exam.id, this.learner.id),
          }));
 ```